File: Source/Policies/HmacPolicy.cs
using HmacManagement.Caching;
using HmacManagement.Headers;

namespace HmacManagement.Policies;

public class HmacPolicy
{
    public readonly string Name;
    public HmacPolicy() => Name = string.Empty;
    public HmacPolicy(string name) => Name = name;
    public KeyCredentials Keys { get; set; } = new();
    public Algorithms Algorithms { get; set; } = new();
    public Nonce Nonce { get; set; } = new();
    public HeaderSchemeCollection HeaderSchemes { get; set; } = new();
}

File: src/Scripts/HmacManager/policies/hmac-policy.ts
import { HeaderSchemeCollection } from "../headers/header-scheme-collection.js";
import { Algorithms } from "./algorithms.js";
import { KeyCredentials } from "./key-credentials.js";
import { Nonce } from "./nonce.js";

export class HmacPolicy {
    readonly name: string;
    readonly keys: KeyCredentials;
    readonly algorithms: Algorithms;
    readonly nonce: Nonce;
    readonly schemes: HeaderSchemeCollection;

    constructor(
        name: string, 
        keys: KeyCredentials, 
        algorithms: Algorithms, 
        nonce: Nonce,
        schemes: HeaderSchemeCollection
    ) {
        this.name = name;
        this.keys = keys;
        this.algorithms = algorithms;
        this.nonce = nonce;
        this.schemes = schemes;
    }
}